ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala

```diff
@@ -130,4 +152,57 @@ object JdbcIndexer {
             )
           }
     )
+
+  private def updatePackageMetadataView(
+      packageStorageBackend: PackageStorageBackend,
+      metrics: Metrics,
+      dbDispatcher: DbDispatcher,
+      packageMetadataView: PackageMetadataView,
+      computationExecutionContext: ExecutionContext,
+      config: IndexerConfig,
+  )(implicit loggingContext: LoggingContext, materializer: Materializer): Future[Unit] = {
+    implicit val ec: ExecutionContext = computationExecutionContext
+    def loadLfArchive(packageId: PackageId): Future[(PackageId, Array[Byte])] =
+      dbDispatcher
+        .executeSql(metrics.daml.index.db.loadArchive)(connection =>
+          packageStorageBackend
+            .lfArchive(packageId)(connection)
+            .getOrElse(
+              // should never happen as we received a reference to packageId
+              sys.error(s"LfArchive does not exist by packageId=$packageId")
+            )
+        )
+        .map(bytes => (packageId, bytes))
+
+    def lfPackagesSource(): Future[Source[PackageId, NotUsed]] =
+      dbDispatcher.executeSql(metrics.daml.index.db.loadPackages)(connection =>
+        Source(packageStorageBackend.lfPackages(connection).keySet)
+      )
+
+    def toMetadataDefinition(packageBytes: Array[Byte]): PackageMetadata =
+      PackageMetadata.from(ArchiveParser.assertFromByteArray(packageBytes))
+
+    def processPackage(archive: (PackageId, Array[Byte])): Future[PackageMetadata] = {
+      val (packageId, packageBytes) = archive
+      Future(toMetadataDefinition(packageBytes)).recover { case NonFatal(e) =>
+        logger.error(s"Failed to decode loaded LF Archive by packageId=$packageId", e)
+        throw e
+      }
+    }
+
+    def onError(): PartialFunction[Throwable, Unit] = { case NonFatal(e) =>
+      logger.error(s"Failed to initialize Package Metadata View", e)
+      throw e
+    }
+
+    Source
+      .futureSource(lfPackagesSource())
+      .mapAsyncUnordered(config.packageMetadataViewLoadParallelism)(loadLfArchive)
+      .mapAsyncUnordered(config.packageMetadataViewProcessParallelism)(processPackage)
+      .runWith(Sink.foreach(packageMetadataView.update))
+      .map(_ => logger.info("Package Metadata View has been initialization"))(
```

Comment:
how about adding a comment to the beginning as well so we have full visibility (on same, info level)?